### DISCUSSION TITLE
fail to launch glsp-client example

### DISCUSSION THREAD

**Discussion by Yunabell-VU** [eclipse-glsp/glsp#1158] [Q&A]
Hello,

I encountered an issue when attempting to launch the example server provided in [glsp-client](https://github.com/eclipse-glsp/glsp-client/tree/master) using the `yarn start:exampleServer` command. It returns an error message that reads as follows:

```shell
yarn run v1.22.19
$ yarn --cwd ./examples/workflow-standalone start:exampleServer
$ yarn ts-node ./scripts/start-example-server.ts
$ /Users/<...>/glsp-client-master/node_modules/.bin/ts-node ./scripts/start-example-server.ts
Check if server executable with version 2.0.0 is present.
Server executable with correct version not found.  Download from npm.
Extract downloaded server tarball
Error: ENOENT: no such file or directory, copyfile '/Users/<...>/glsp-client-master/examples/workflow-standalone/server/package/wf-glsp-server-node.js' -> '/Users/<...>/glsp-client-master/examples/workflow-standalone/server/workflow-server-2.0.0.js'
    at Object.copyFileSync (node:fs:2866:3)
    at downloadIfNecessary (/Users/<...>/glsp-client-master/examples/workflow-standalone/scripts/start-example-server.ts:63:8)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async run (/Users/<...>/glsp-client-master/examples/workflow-standalone/scripts/start-example-server.ts:24:24) {
  errno: -2,
  syscall: 'copyfile',
  code: 'ENOENT',
  path: '/Users/<...>/glsp-client-master/examples/workflow-standalone/server/package/wf-glsp-server-node.js',
  dest: '/Users/<...>/glsp-client-master/examples/workflow-standalone/server/workflow-server-2.0.0.js'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
It appears to be failing to fetch the server from the script. Could you please advise on how to resolve this issue, or alternatively, is it possible to run the server from [glsp-server](https://github.com/eclipse-glsp/glsp-server) while still using the standalone client example?

Additionally, I have a question about integrating glsp into a React or Svelte project. Is this a viable option, and if so, what are the steps to incorporate the glsp-client framework into a front-end framework? I would greatly appreciate any guidance you can provide on this matter.

Thanks!

**Comment by tortmayr** [comment]
Hi @Yunabell-VU,

thanks for reaching out. As it turns out there was a small bug in the script that should start the example server.
This has been fixed now. You can either use the patch PR of [glsp-client](https://github.com/eclipse-glsp/glsp-client/pull/302) or the tagged [2.0. release](https://github.com/eclipse-glsp/glsp-client/tree/v2.0.0).

>. Could you please advise on how to resolve this issue, or alternatively, is it possible to run the server from [glsp-server](https://github.com/eclipse-glsp/glsp-server) while still using the standalone client example?

Yes that is possible. You can either use the node/TS or Java server vairant:
- [Java](https://github.com/eclipse-glsp/glsp-server):
   -  Checkout the repo and build with:
    ```bash
    mvn clean verify -Pm2 -Pfatjar
    ```
   -  Then state the example jar with in websocket mode with:
     ```bash
      cd examples/org.eclipse.glsp.example.workflow/target
      java -jar org.eclipse.glsp.example.workflow-X.X.X-SNAPSHOT-glsp.jar
     ```
 - [Node/TS](https://github.com/eclipse-glsp/glsp-server-node)
   - Checkout the repo and build with:
    ```bash
     yarn
    ```
    - Start the server in websocket mode with:
    ```bash
    yarn start:websocket
    ```
 
 Once the server is running, open the glsp-client/examples/workflow-standalone/app/diagram.html file in your favorite browser.
 
   >Additionally, I have a question about integrating glsp into a React or Svelte project. Is this a viable option, and if so, what are the steps to incorporate the glsp-client framework into a front-end framework? I would greatly appreciate any guidance you can provide on this matter.
 
 Unfortunately I'm not aware of any adopters that have tried to integrate GLSP into a Svelte or React project, so I can not really provide you with any guidance on this matter. @planger @martin-fleck-at Does anyone of you have experience with integrating GLSP into react or svelte?
